src/hub/vendor/abstract.py

```diff
@@ -47,3 +53,19 @@ def run(self) -> None:
 
     def create_data(self, **kwargs) -> Dict[str, str]:
         return dict(event_id=self.payload.id, event_type=self.payload.type, **kwargs)
+
+    def customer_event_to_all_routes(self, data_projection, data):
+        subsets = []
+        for route in data_projection:
+            try:
+                logger.debug("sending to", key=route)
+                subset = dict((k, data[k]) for k in data_projection[route] if k in data)
+                payload = {"type": route, "data": json.dumps(subset)}
+                subsets.append(payload)
+                logger.info("subset", subset=subset)
+                logger.debug("sent to", key=route)
+            except Exception as e:
+                # log something and maybe change the exception type.
+                logger.error("projection exception", error=e)
+        else:
```

Comment:
This `else` doesn't match up to anything and should be removed.

Reply:
I disagree, I purposefully put this in the else block in the for loop.  it is in the correct place.

Reply:
I thought that the `else` was only used if there was a `break` in a `for` loop?

Reply:
Else will only be used if there is NOT a break in the for loop!

Reply:
https://book.pythontips.com/en/latest/for_-_else.html#else-clause